utils/writeable/src/fallible.rs

```diff
@@ -0,0 +1,660 @@
+    ///
+    /// # Examples
+    ///
+    /// Successful behavior on `Option<T>`, which implements [`TryWriteable`]:
+    ///
+    /// ```
+    /// use writeable::TryWriteable;
+    ///
+    /// let try_writeable = Some("example");
+    /// assert_eq!(
+    ///     try_writeable.checked().try_write_to_string().unwrap(),
+    ///     "example"
+    /// );
+    /// ```
+    ///
+    /// Failure behavior:
+    ///
+    /// ```
+    /// use writeable::TryWriteable;
+    ///
+    /// let try_writeable = None::<&str>;
+    /// assert!(matches!(
+    ///     try_writeable.checked().try_write_to_string(),
+    ///     Err(_),
+    /// ));
+    /// ```
+    #[inline]
+    fn checked(&self) -> CheckedWriteable<&Self> {
+        CheckedWriteable(self)
+    }
+    /// Handle errors by writing a replacement to the output string.
+    ///
+    /// # Examples
+    ///
+    /// Successful behavior on `Option<T>`, which implements [`TryWriteable`]:
+    ///
+    /// ```
+    /// use writeable::TryWriteable;
+    ///
+    /// let try_writeable = Some("example");
+    /// writeable::assert_writeable_eq!(try_writeable.lossy(), "example");
+    /// ```
+    ///
+    /// Failure behavior:
+    ///
+    /// ```
+    /// use writeable::TryWriteable;
+    ///
+    /// let try_writeable = None::<&str>;
+    /// writeable::assert_writeable_eq!(try_writeable.lossy(), "�");
+    /// ```
+    #[inline]
+    fn lossy(&self) -> LossyWriteable<&Self> {
+        LossyWriteable(self)
+    }
+    /// Handle errors by panicking.
+    ///
+    /// # Examples
+    ///
+    /// Successful behavior on `Option<T>`, which implements [`TryWriteable`]:
+    ///
+    /// ```
+    /// use writeable::TryWriteable;
+    ///
+    /// let try_writeable = Some("example");
+    /// writeable::assert_writeable_eq!(try_writeable.panicky(), "example");
+    /// ```
+    ///
+    /// Failure behavior:
+    ///
+    /// ```should_panic
+    /// use writeable::TryWriteable;
+    /// use writeable::Writeable;
+    ///
+    /// let try_writeable = None::<&str>;
+    /// try_writeable.panicky().write_to_string();
+    /// ```
+    #[inline]
+    fn panicky(&self) -> PanickyWriteable<&Self> {
+        PanickyWriteable(self)
+    }
+    /// Handle errors by panicking in debug mode and writing a replacement in release mode.
+    ///
+    /// # Examples
+    ///
+    /// Successful behavior on `Option<T>`, which implements [`TryWriteable`]:
+    ///
+    /// ```
+    /// use writeable::TryWriteable;
+    ///
+    /// let try_writeable = Some("example");
+    /// writeable::assert_writeable_eq!(try_writeable.gigo(), "example");
+    /// ```
+    ///
+    /// Failure behavior:
+    ///
+    /// ```
+    /// use writeable::TryWriteable;
+    ///
+    /// let try_writeable = None::<&str>;
+    /// if cfg!(not(debug_assertions)) {
+    ///     writeable::assert_writeable_eq!(try_writeable.gigo(), "�");
+    /// }
+    /// ```
+    #[inline]
+    fn gigo(&self) -> GigoWriteable<&Self> {
```

Comment:
hmm, so we haven't been using the term "gigo" on our API before. I think it's not one of the universally known acronyms, so I'd like to find a better name for it.